### PR TITLE
rancher2-deployment/improve-docs

### PR DESCRIPTION
**Pull request by bemeyert**
improve documentation

### CODE REVIEW

**Comment by bemeyert** [comment]
@falsanu Please review and let me know if this meets your expectations for the "depth" of documentation.

**Comment by bemeyert** [comment]
@falsanu Can you please review my changes?

File: rancher2-deployment/Dockerfile [hunk 1]
```diff
@@ -1,23 +1,57 @@
...
+ENV DEPLOYMENT_TEMPLATE_FILE="deployment.tmpl.yml"
+
+# Sets the rancher context containing of <cluster-id>:<project-id>.
 ENV RANCHER_CONTEXT=""
 
-# Mandatory project name:
-ENV PROJECT=""
+# Sets the project name. Must be the name of the Rancher2 project. The value
+# is used together with the value of ENVIRONMENT to construct the name of the
+# K8s namespace. Can be overriden by PROJECT.
```
Comment (falsanu) [inline rancher2-deployment/Dockerfile]: where is PROJECT?

File: rancher2-deployment/Dockerfile [hunk 2]
```diff
@@ -1,23 +1,57 @@
 FROM rancher/cli2 AS release
-ARG DEPLOYMENT_TEMPLATE_FILE="deployment.tmpl.yml"
-ENV DEPLOYMENT_TEMPLATE_FILE=${DEPLOYMENT_TEMPLATE_FILE}
 
-# Mandatory rancher context:
+# This image's main purpose is to run deployments to a Rancher2 cluster. It
+# leverages the rancher2-cli, kubectl and gomplate to accomplish that.
```
Comment (falsanu) [inline rancher2-deployment/Dockerfile]: what is gomplate?

File: rancher2-deployment/entrypoint.sh [hunk 3]
```diff
@@ -3,30 +3,40 @@
...
+# - IFS: word splitting is done by <newline> and <tab>, WITHOUT <space>.
+# For more details s. `help set`, `man bash | less -p '   IFS '` and
+# http://redsymbol.net/articles/unofficial-bash-strict-mode/
 set -euo pipefail
 IFS=$'\n\t'
 
 ###############################################################################
 ### define/initialize script wide variables
 
+# This programs name without any leading slashes (i.e same as basename).
```
Comment (falsanu) [inline rancher2-deployment/entrypoint.sh]: I don't get it

File: rancher2-deployment/entrypoint.sh [hunk 4]
```diff
@@ -3,30 +3,40 @@
...
 PROGNAME="${0##*/}"
+# Name of the resulting K8s YAML file which gomplate will create.
 OUTPUT_FILE="./deployment.yml"
-# Make a dry run
+# If set, make a dry run
```
Comment (falsanu) [inline rancher2-deployment/entrypoint.sh]: where do i set dry run?

File: rancher2-deployment/Dockerfile (hunk 1 above)
Comment (kr1sp1n) [reply rancher2-deployment/Dockerfile]: The `PROJECT` env var must be set in the specific project repo. Usually this is a constant value in the `Makefile` in the root of a project repo.

File: rancher2-deployment/Dockerfile (hunk 2 above)
Comment (kr1sp1n) [reply rancher2-deployment/Dockerfile]: A flexible commandline tool for template rendering.  https://github.com/hairyhenderson/gomplate

File: rancher2-deployment/entrypoint.sh (hunk 4 above)
Comment (kr1sp1n) [reply rancher2-deployment/entrypoint.sh]: You control all the behaviour with environment variables. So set `DRY_RUN` e.g. in your `.env` file in the specific project repo.

File: rancher2-deployment/Dockerfile (hunk 2 above)
Comment (bemeyert) [reply rancher2-deployment/Dockerfile]: This is explained in README.md. Thus I don't see a reason to further describe it here.

File: rancher2-deployment/entrypoint.sh (hunk 4 above)
Comment (bemeyert) [reply rancher2-deployment/entrypoint.sh]: This is explained in the README.md. I don't see why this should be repeated here.

File: rancher2-deployment/Dockerfile (hunk 1 above)
Comment (bemeyert) [reply rancher2-deployment/Dockerfile]: I changed some to make this less confusing.

File: rancher2-deployment/entrypoint.sh (hunk 3 above)
Comment (bemeyert) [reply rancher2-deployment/entrypoint.sh]: I added more text which should explain things better

**Comment by bemeyert** [comment]
@falsanu I made some changes. Please take a look.

**Comment by bemeyert** [comment]
Meanwhile I try to fix the conflicts. This needs some extensive fiddling... *grmbl*

**Review by falsanu — APPROVED** [review]

**Pull request merged**